astropy/coordinates/baseframe.py

```diff
@@ -530,7 +535,13 @@ def realize_frame(self, representation):
             A new object with the same frame attributes as this one, but
             with the ``representation`` as the data.
         """
-        return self._apply('replicate', _framedata=representation)
+        # Here we pass representation=None to _apply, since we do not want
+        # to insist that the realized frame has the same representation as
+        # self.  [Avoids breaking sunpy; see gh-6208]
+        # TODO: expose this? But unhandy change of name, since in the
```

Comment:
In #6219 I added `representation_cls` and am planning to deprecate `representation` - I guess you could use `_representation_cls` just to be consistent? I don't really care.